### PR TITLE
OTC-878: add permissions to all gql mutations and queries

### PR DESCRIPTION
**Pull request by jdolkowski**
https://openimis.atlassian.net/browse/OTC-878

### CODE REVIEW

File: self_registration/gql_mutations.py [hunk 1]
```diff
@@ -42,10 +45,14 @@ class Meta:
     edge_count = Int()
 
     def resolve_total_count(root, info, **kwargs):
+        if not info.context.user.is_authenticated:
+            raise PermissionDenied(_("unauthorized"))
         print(root.length)
```
Comment (dragos-dobre) [inline self_registration/gql_mutations.py]: Is this for debug? Please remove if not need anymore.

File: self_registration/gql_mutations.py [hunk 2]
```diff
@@ -77,7 +84,12 @@ def mutate(self, info, file, insureeCHFID, email, phone):
         print(insuree_obj.pk)
```
Comment (dragos-dobre) [inline self_registration/gql_mutations.py]: Is this for debug? Please remove if not need anymore.

File: self_registration/gql_mutations.py [hunk 3]
```diff
@@ -142,6 +156,8 @@ class Arguments:
 
     @classmethod
     def mutate(cls, root, info, **kwargs):
+        if not info.context.user.has_perms(SelfRegistrationConfig.gql_mutation_add_feedback_perms):
+            raise PermissionDenied(_("unauthorized"))
         print(kwargs)
```
Comment (dragos-dobre) [inline self_registration/gql_mutations.py]: Is this for debug? Please remove if not needed anymore.

File: self_registration/gql_mutations.py [hunk 4]
```diff
@@ -158,6 +174,8 @@ class Input:
 
     @classmethod
     def mutate_and_get_payload(cls, root, info, **input):
+        if not info.context.user.has_perms(SelfRegistrationConfig.gql_mutation_add_notification_perms):
+            raise PermissionDenied(_("unauthorized"))
         print('CreateNoticeMutation mutate')
```
Comment (dragos-dobre) [inline self_registration/gql_mutations.py]: Is this for debug? Please remove if not needed anymore.

File: self_registration/gql_mutations.py [hunk 5]
```diff
@@ -180,6 +198,8 @@ class Input(OpenIMISMutation.Input):
 
     @classmethod
     def mutate_and_get_payload(cls, root, info, **input):
+        if not info.context.user.has_perms(SelfRegistrationConfig.gql_mutation_update_notification_perms):
+            raise PermissionDenied(_("unauthorized"))
         dfprint('UpdateNoticeMutation mutate')
```
Comment (dragos-dobre) [inline self_registration/gql_mutations.py]: Is this for debug? Please remove if not needed anymore.

File: self_registration/gql_mutations.py [hunk 6]
```diff
@@ -224,6 +246,8 @@ class Arguments:
 
     @classmethod
     def mutate(self, info, cls, **kwargs):
+        if not info.context.user.has_perms(SelfRegistrationConfig.gql_mutation_add_insuree_reg_perms):
+            raise PermissionDenied(_("unauthorized"))
         dfprint(kwargs)
```
Comment (dragos-dobre) [inline self_registration/gql_mutations.py]: Is this for debug? Please remove if not needed anymore.

File: self_registration/gql_mutations.py [hunk 7]
```diff
@@ -398,6 +422,8 @@ class Arguments:
 
     @classmethod
     def mutate(self, info, cls, **kwargs):
+        if not info.context.user.has_perms(SelfRegistrationConfig.gql_mutation_add_insuree_perms):
+            raise PermissionDenied(_("unauthorized"))
         dfprint('CreateInsureeMutation mutate')
```
Comment (dragos-dobre) [inline self_registration/gql_mutations.py]: Is this for debug? Please remove if not needed anymore.

File: self_registration/schema.py [hunk 8]
```diff
@@ -365,15 +404,21 @@ def generate_token(self):
...
     def resolve_validate_insuree(self, info, phone_number):
+        if not info.context.user.has_perms(SelfRegistrationConfig.gql_query_insuree_perms):
+            raise PermissionDenied(_("unauthorized"))
         jot = InsureeTempReg.objects.filter(phone_number=phone_number).first() #(Q(phone_no=phone_number | Q(card_id=card_id))):
         return jot
     
     def resolve_track_registration_status(self, info, phone_no):
+        if not info.context.user.has_perms(SelfRegistrationConfig.gql_query_insuree_perms):
+            raise PermissionDenied(_("unauthorized"))
         print('asdasd',phone_no)
```
Comment (dragos-dobre) [inline self_registration/schema.py]: Is this for debug? Please remove if not needed anymore.

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

**Comment by jdolkowski** [comment]
@dragos-dobre I removed all of the prints. I left them because I didn't want to change the code I didn't write, but I guess they do not serve any purpose whatsoever.

**Review by delcroip — APPROVED** [review]

**Comment by delcroip** [comment]
@dragos-dobre  please merge it when happy with the changes

**Pull request merged**